src/LibHdb++MySQL.cpp

```diff
@@ -1490,368 +1535,334 @@ void HdbPPMySQL::event_Attr(string name, unsigned char event)
-				value_data[1]=(Type)0;	//useless
-			}
-			else
-			{
-				is_null[1]=0;
-				value_data[1] = value_w[0];
-			}
-		}
+		ostringstream query_str;
+		if(!ignoreduplicates)
+			query_str << "INSERT INTO ";
 		else
-		{
-			is_null[1]=1;
-			value_data[1]=(Type)0;	//useless
-		}
-
-		int_data[0] = ID;
-		if(detected_quality)
-			int_data[1] = quality;
-		double_data[0] = ev_time;
+			query_str << "INSERT IGNORE INTO ";
+		query_str << m_dbname << "." << table_name <<
+				" (" << SC_COL_ID << "," << SC_COL_EV_TIME << ",";
+		if(detected_insert_time)
+			query_str << SC_COL_INS_TIME << ",";
 		if(detected_recv_time)
-			double_data[1] = rcv_time;
+			query_str << SC_COL_RCV_TIME << ",";
+		if(detected_quality)
+			query_str << SC_COL_QUALITY << ",";
 		if(detected_error)
-		{
-			int_data[2] = ERR_ID;
-			if(ERR_ID < 0)
-				is_null[2]=1;
-			else
-				is_null[2]=0;
-		}
-		memset(plog_bind, 0, sizeof(plog_bind));
-
-		size_t plog_bind_ind=0;
-
-		plog_bind[plog_bind_ind].buffer_type= MYSQL_TYPE_LONG;
-		plog_bind[plog_bind_ind].buffer= (void *)&int_data[0];
-		plog_bind[plog_bind_ind].is_null= 0;
-		plog_bind[plog_bind_ind].length= 0;
-		plog_bind_ind++;
+			query_str << SC_COL_ERROR_DESC_ID << ",";
+		query_str << SC_COL_VALUE_R;
+		if(!(write_type == Tango::READ))	//RW
+			query_str << "," << SC_COL_VALUE_W;
+		query_str << ")";
 
-		plog_bind[plog_bind_ind].buffer_type= MYSQL_TYPE_DOUBLE;
-		plog_bind[plog_bind_ind].buffer= (void *)&double_data[0];
-		plog_bind[plog_bind_ind].is_null= 0;
-		plog_bind[plog_bind_ind].length= 0;
-		plog_bind_ind++;
+		query_str << " VALUES";
 
-		if(detected_recv_time)
+		for(size_t idx=0; idx < insert_size; ++idx)
 		{
-			plog_bind[plog_bind_ind].buffer_type= MYSQL_TYPE_DOUBLE;
-			plog_bind[plog_bind_ind].buffer= (void *)&double_data[1];
-			plog_bind[plog_bind_ind].is_null= 0;
-			plog_bind[plog_bind_ind].length= 0;
-			plog_bind_ind++;
-		}
-		if(detected_quality)
-		{
-			plog_bind[plog_bind_ind].buffer_type= MYSQL_TYPE_LONG;
-			plog_bind[plog_bind_ind].buffer= (void *)&int_data[1];
-			plog_bind[plog_bind_ind].is_null= 0;
-			plog_bind[plog_bind_ind].length= 0;
-			plog_bind_ind++;
+			query_str << " (?,FROM_UNIXTIME(?),";
+			if(detected_insert_time)
+				query_str << "NOW(6),";//insert_time
+			if(detected_recv_time)
+				query_str << "FROM_UNIXTIME(?),";//recv_time
+			if(detected_quality)
+				query_str << "?,";//quality
+			if(detected_error)
+				query_str << "?,";//error
+			query_str << "?";	//value_r
+			if(!(write_type == Tango::READ))	//RW
+				query_str << ",?";//value_w
+			query_str << ")";
+			if(idx < insert_size-1)
+				query_str << ",";
 		}
 
+		int param_count_single = 3;	//param in single value insert
+		if(detected_recv_time)
+			param_count_single++;
+		if(detected_quality)
+			param_count_single++;
 		if(detected_error)
+			param_count_single++;
+		if(write_type != Tango::READ)
+			param_count_single ++;
+		int param_count = param_count_single*insert_size;
+		uint32_t retry_cnt=0;
```

Comment:
Same as for batch_size, what's the need of uint32_t instead size_t or unsigned int ?